tests/orm/nodes/data/test_data.py

```diff
@@ -109,6 +109,25 @@ def _generate_class_instance(data_class):
             )
             return instance
 
+        if (
+            data_class is orm.AbstractCode
+            or data_class is orm.Code
+            or data_class is orm.InstalledCode
+            or data_class is orm.ContainerizedCode
+            or data_class is orm.PortableCode
+        ):
+            # Taken from `test_code.py`
+            import uuid
+
+            computer = orm.Computer(
+                label=uuid.uuid4().hex,
+                transport_type='core.local',
+                hostname='localhost',
+                scheduler_type='core.slurm',
+            ).store()
+            instance = orm.Code(remote_computer_exec=(computer, '/bin/cat'))
```

Comment:
By always instantiating `orm.Code`, you are not really testing the other subclasses. Would be best to really instantiate each class separately, no?

Reply:
Fair enough, this was just a lazy shortcut I have to admit. Working on implementing it properly. Stumbled on #6568 while doing so.

Reply:
Fine to add that property in this PR if you want.

Reply:
On it!

Reply:
Implemented it now. I added some comments to commit 902d10f, pasted here again for convenience:

Some notes on the implementation:
- Removed the `put_object_from_tree` from the `__init__`, as this is
  done in the `setter` which is anyway called when the `property` is
  actually set in the `__init__`.
- Added check for absolute Path (would otherwise be
captured in `Repository.put_object_from_tree`, but maybe better to catch
here already).
- Not sure if the repository should be erased before putting
another `object_from_tree`. The logic I followed is that if one wants to
set another `filepath_files` to not keep the previous ones and get a
polluted repository, but rather just setting the new one. However, it
should be avoided ofc that unwanted things are removed from the
repository. And, if one wants to add multiple directories, etc., this
situation is described in the constructor docstring, and one can use the
methods of the `NodeRepository`, so I guess this shouldn't be done via
setting the `filepath_files`

Mainly the last point, I'm not sure if this is a good idea. Apart from that, I saw some of your previous PRs on the `PortableCode`, e.g., #5510 and #6406, so tried to be consistent with the implementation as is.